### ISSUE TITLE
Adding model support to Neural Network model

### ISSUE THREAD

**Issue by laramdemajo** [viadee/anchorsOnR#66]
Hi, I am trying to apply Anchors on the neural network model I am working with. I tried following the details in the models.R file however did not manage to understand what I really have to do. Can you kindly provide an actual example of how model support can be achieved please?

This is what I am doing:

```
model_anchors = anchors::as_classifier(NNet, labels=factor(c(1,0)))
anchor_explainer <- anchors(trainset, model_anchors, target="loan_status")
anchor_explanations <- anchors::explain(validset, anchor_explainer)
printExplanations (anchor_explainer, anchor_explanations)
```

However, when it gets to the third line I am getting below error:

> Error in predict_model.default(explainer$model, x) : argument "type" is missing, with no default

I am also working with LIME library provided at https://github.com/thomasp85/lime and using as_classifier works just fine.

**Comment by thllwg** [comment]
Hi @laramdemajo and thank you for opening this issue!
You're right, `as_classifier` does not yield the expected result. The reason for this is that the evaluated type was not passed on correctly within anchors. This is fixed now but I'm still working on an exemplary explanation of a neural network with anchors. I'll update the issue again once I'm done!

**Comment by thllwg** [comment]
I have added an example of a keras-based neural network (`inst\examples\keras_iris.R`). Please note that this example is rather complex in comparison to the machine learning examples integrated so far. 

To extend anchors in order to support a neural network that is not keras-based, provide your own `predict_model` and `model_type` method. In the classification context, this method should return either a vector of predictions (e.g. `c("healthy", "healthy", "sick", "sick", "healthy")` or a one-hot encoded matrix of predictions. You may supply labels for them as you did in your code snippet. 

Please feel free to share a minimal (not) working example, i.e. including information about the neural network you've trained.

**Comment by thllwg** [comment]
I consider this issue solved. If you have any other questions regarding this or other aspects of anchors please don't hesitate to comment or open another issue! :)
